wagtail/admin/views/scheduled_pages.py

```diff
@@ -0,0 +1,110 @@
+from datetime import timedelta
+
+from django.conf import settings
+from django.core.exceptions import PermissionDenied
+from django.shortcuts import get_object_or_404, redirect
+from django.template.response import TemplateResponse
+from django.utils import timezone
+from django.utils.http import url_has_allowed_host_and_scheme
+from django.utils.translation import gettext_lazy as _
+
+from wagtail.admin import messages
+from wagtail.admin.ui.components import Component
+from wagtail.core.models import Page, UserPagePermissionsProxy
+
+
+def get_scheduled_pages_for_user(request):
+    user_perms = UserPagePermissionsProxy(request.user)
+    pages = (
+        Page.objects.annotate_approved_schedule()
+        .filter(_approved_schedule=True)
+        .prefetch_related("content_type")
+        .order_by("-first_published_at")
+        & user_perms.publishable_pages()
+    )
+
+    if getattr(settings, "WAGTAIL_I18N_ENABLED", False):
+        pages = pages.select_related("locale")
+    return pages
+
+
+class ScheduledPagesPanel(Component):
+    name = "scheduled_pages"
+    template_name = "wagtailadmin/scheduled_pages/panel.html"
+    order = 200
+
+    def get_context_data(self, parent_context):
+        request = parent_context["request"]
+        context = super().get_context_data(parent_context)
+        context["pages_to_be_scheduled"] = get_scheduled_pages_for_user(request)
+        context["request"] = request
+        context["csrf_token"] = parent_context["csrf_token"]
+        return context
+
+
+def publish(request, page_id):
+    page = get_object_or_404(Page, id=page_id).specific
+    if not page.permissions_for_user(request.user).can_publish():
+        raise PermissionDenied
+
+    new_go_live_timestamp = timezone.now() - timedelta(seconds=1)
+    page.go_live_at = new_go_live_timestamp
+    page.save()
+
+    # Save revision
+    revision = page.save_revision(user=request.user, log_action=True)
+    revision.publish()
```

Comment:
the `published_scheduled_page` management command does not save a new revision, but publishes the revision that was scheduled. We should be consistent (https://github.com/wagtail/wagtail/blob/main/wagtail/management/commands/publish_scheduled.py#L123-L125 and https://github.com/wagtail/wagtail/blob/main/wagtail/management/commands/publish_scheduled.py#L149)

Reply:
as before, the management command only works on revisions that have schedules in the past. So you can't really call publish on a scheduled revision right now, because it will be ignored. So unless the core publish mechanism is changed to allow publishing a revision scheduled in the future, this is the way to go.